jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray.c

```diff
@@ -55,13 +56,130 @@ ecma_builtin_typedarray_from (ecma_value_t this_arg, /**< 'this' argument */
                               const ecma_value_t *arguments_list_p, /**< arguments list */
                               ecma_length_t arguments_list_len) /**< number of arguments */
 {
-  JERRY_UNUSED (this_arg);
-  JERRY_UNUSED (arguments_list_p);
-  JERRY_UNUSED (arguments_list_len);
+  JERRY_ASSERT (arguments_list_len == 0 || arguments_list_p != NULL);
 
-  /* TODO: inplement 'from' */
+  if (!ecma_is_constructor (this_arg))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("this is not a constructor."));
+  }
+
+  ecma_value_t source;
+  ecma_value_t map_fn = ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED);
+  ecma_value_t this_in_fn = ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED);
+
+  if (arguments_list_len == 0)
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("source is missing"));
+  }
+  else
```

Comment:
unnecessary `else` because of the `return` statement.

```c
if (arguments_list_len == 0)
{
  return ecma_raise_type_error (ECMA_ERR_MSG ("source is missing"));
}

source = arguments_list_p[0];
```